packages/veraswap-sdk/src/calls/getBridgeSwapCalls.ts

```diff
@@ -27,11 +56,202 @@ export interface GetBridgeSwapCallsParams extends GetCallsParams {
+        remoteERC7579ExecutorRouter,
+        remoteSwapParams,
+        tokenStandard,
+        approveAmount,
+        permit2,
+    } = params;
+    const { initData, salt } = params.createAccount;
+    const contracts = params.contracts ?? {
+        execute: LOCAL_KERNEL_CONTRACTS.execute,
+        ownableSignatureExecutor: LOCAL_KERNEL_CONTRACTS.ownableSignatureExecutor,
+    };
+
+    if (!(tokenStandard === "HypERC20" || tokenStandard === "HypERC20Collateral" || tokenStandard === "NativeToken")) {
+        throw new Error(`Unsupported standard for bridge and swap: ${tokenStandard}`);
+    }
+
+    // Create account if needed
+    const createAccountCalls = await getKernelFactoryCreateAccountCalls(queryClient, wagmiConfig, {
+        chainId,
+        account,
+        ...params.createAccount,
+    });
+    const kernelAddress = createAccountCalls.kernelAddress;
+
+    let transferRemoteCalls: GetCallsReturnType;
+    if (tokenStandard === "NativeToken") {
+        // Assume that if the token is native, we are using the Orbiter bridge
+        const orbiterCall = getOrbiterETHTransferTransaction({
+            recipient: kernelAddress,
+            amount,
+            ...params.orbiterParams!,
+        });
+
+        transferRemoteCalls = { calls: [{ ...orbiterCall, account: kernelAddress }] };
+    } else {
+        // TODO: handle future case where we bridge USDC with orbiter
+
+        // Encode transferRemote calls, pull funds from account if needed
+        transferRemoteCalls = await getTransferRemoteWithFunderCalls(queryClient, wagmiConfig, {
+            chainId,
+            token,
+            tokenStandard,
+            account: kernelAddress,
+            funder: account,
+            destination,
+            recipient: kernelAddress,
+            amount,
+            hookMetadata,
+            hook,
+            approveAmount,
+            permit2,
+        });
+    }
+
+    // From here refactor into a function
+    const swapParams: GetSwapCallsParams = {
+        account: kernelAddress,
+        chainId: destination,
+        approveExpiration: remoteSwapParams.approveExpiration ?? "MAX_UINT_48",
+        ...remoteSwapParams,
+    };
+
+    const { calls: swapRemoteCalls } = await getSwapCalls(queryClient, wagmiConfig, swapParams);
+
+    const remoteCallData = encodeCallArgsBatch(swapRemoteCalls);
+
+    await switchChain(wagmiConfig, { chainId: destination });
+
+    // @ts-expect-error  queryKeyHashFn should exist and is needed to serialize a bigint
+    const nonce = await queryClient.fetchQuery({
+        ...readContractQueryOptions(wagmiConfig, {
+            chainId: destination,
+            address: LOCAL_KERNEL_CONTRACTS.ownableSignatureExecutor,
+            abi: OwnableSignatureExecutor.abi,
+            functionName: "getNonce",
+            args: [kernelAddress, 0n],
+        }),
+        queryKeyHashFn: hashFn,
+    });
+
+    // Sign execution data
+    const signatureExecution = {
```

Comment:
signature execution should only be needed on deployment

Reply:
ok, better to refactor this part to just use getOwnableExecutorExecuteCalls